packages/babel-plugin-proposal-pipeline-operator/src/hackVisitor.ts

```diff
@@ -1,8 +1,27 @@
 import { types as t } from "@babel/core";
+import type { NodePath, Visitor } from "@babel/traverse";
 
-const topicReferenceReplacementVisitor = {
-  TopicReference(path) {
-    path.replaceWith(t.cloneNode(this.topicVariable));
+const topicReferenceVisitor: Visitor<{
+  topicReferences: NodePath<t.TopicReference>[];
+  sideEffectsBeforeFirstTopicReference: boolean;
+}> = {
+  exit(path, state) {
+    if (path.isTopicReference()) {
+      state.topicReferences.push(path);
+    } else {
+      if (
+        state.topicReferences.length === 0 &&
+        !state.sideEffectsBeforeFirstTopicReference &&
+        !path.isPure()
+      ) {
+        state.sideEffectsBeforeFirstTopicReference = true;
```

Comment:
Is sideEffectsBeforeFirstTopicReference invalidated only when side effects occur before the first topic reference in an RHS? For example, will it be set to true for `x |> (f(x), g(x))`?

Reply:
I assume you mean `x |> (f(#), g(#))`. It will not be set to true because there are no side effects before the first `#`. However, since there are multiple `#`s we won't inline anything.